src/inmanta/agent/executor.py

```diff
@@ -631,6 +631,7 @@ class DeployReport:
     messages: list[LogLine]
     changes: dict[str, AttributeStateChange]
     change: Optional[Change]
+    report: Optional[bool] = False
```

Comment:
Could we extend `Change` for this instead? It seems appropriate to me that we would make it mutually exclusive with the other `Change` modes, and it takes away an ambiguous field.

Or we can simply rely on the `resource_state`. I'm not quite sure if we want to use `change`, `resource_state` or both. Perhaps just `resource_state`?